### ISSUE TITLE
(Fedora) macros.meson changes prevent me from building subproject

### ISSUE THREAD

**Issue by eclipseo** [mesonbuild/meson#7401]
The somewhat recent change to macros.meson from ninja to meson compile makes my SPEC not compile anymore (on Fedora Rawhide). I am not sure how I am supposed to fix it with the new macros.

Commit: https://github.com/mesonbuild/meson/commit/0a61f511aa1960ac0d3f9b5e50e35f5f603b99b7#diff-4c5c803be0ad7acb72578be0880b3a35

My SPEC before was doing:

```
%build
%meson
%meson_build
%meson_build doc/html
```

which was translated to:

`/usr/bin/ninja -v -j8 -C x86_64-redhat-linux-gnu doc/html`

Now it is translated as:

`/usr/bin/meson compile -C x86_64-redhat-linux-gnu -j 8 --verbose doc/html`

which is not a valid command.

I am not super familiar with meson commands, how am I supposed to build a different target with meson compile?

Note:
doc/html is a target not compiled by default so I need to build it independently:
```
    custom_target('doc',
                  build_by_default: false,
                  command: [doxygen, doxyfile],
                  output: ['html']
    )
```
https://code.videolan.org/videolan/dav1d/-/blob/master/doc/meson.build

**Comment by eclipseo** [comment]
Of course I can:

`%ninja_build -C %{_vpath_builddir} doc/html`

but I'm wondering if there is a Meson way of doing that instead.

**Comment by jpakkane** [comment]
Ping @ignatenkobrain.

**Comment by ignatenkobrain** [comment]
@jpakkane sadly nothing I can do here. I can't find how `meson compile` can run some specific targets. If there is no way and ninja should be used directly, @eclipseo knows how to do that :)

**Comment by TheQwertiest** [comment]
This macro is targeting meson from the master branch (a future 0.55.0 release), which *has* a target support in `meson compile`. 

If you're using a different version of meson, then you should take a macro from a relevant branch instead (e.g. `0.54`).

**Comment by ignatenkobrain** [comment]
Oh, so it is my fault. Can you link me to the relevant change in master?

**Comment by ignatenkobrain** [comment]
I guess it is 5696a5abbaaff75279d9c50d431de47f35dc6228

**Comment by TheQwertiest** [comment]
@ignatenkobrain , why is it your fault? =)
Or do you mean that this macro was uploaded\used somewhere by you as-is? 

PS: That's not the only commit that is relevant. You need at least your PR for `--verbose` as well.

**Comment by ignatenkobrain** [comment]
> Or do you mean that this macro was uploaded\used somewhere by you as-is?

Yes, I backported it to Fedora package.

> PS: That's not the only commit that is relevant. You need at least your PR for --verbose support as well.

I did that, but I did not notice that 0.54 does not support specifying target.

PS: That's not the only commit that is relevant. You need at least your PR for --verbose support as well.

**Comment by TheQwertiest** [comment]
TLDR: this is not a meson problem

**Comment by ignatenkobrain** [comment]
Agree, I'll backport necessary patches in Fedora.
